packages/components/src/toggle-group-control/toggle-group-control/component.tsx

```diff
@@ -20,43 +20,75 @@ import { VisualLabelWrapper } from './styles';
 import * as styles from './styles';
 import { ToggleGroupControlAsRadioGroup } from './as-radio-group';
 import { ToggleGroupControlAsButtonGroup } from './as-button-group';
+import type { ElementOffsetRect } from '../../utils/element-rect';
 import { useTrackElementOffsetRect } from '../../utils/element-rect';
 import { useOnValueUpdate } from '../../utils/hooks/use-on-value-update';
 import { useEvent, useMergeRefs } from '@wordpress/compose';
 
 /**
- * A utility used to animate something (e.g. an indicator for the selected option
- * of a component).
+ * A utility used to animate something in a container component based on the "offset
+ * rect" (position relative to the container and size) of a subelement. For example,
+ * this is useful to render an indicator for the selected option of a component, and
+ * to animate it when the selected option changes.
  *
- * It works by tracking the position and size (i.e., the "rect") of a given subelement,
- * typically the one that corresponds to the selected option, relative to its offset
- * parent. Then it:
+ * Takes in a container element and the up-to-date "offset rect" of the target
+ * subelement, obtained with `useTrackElementOffsetRect`. Then it does the following:
  *
- * - Keeps CSS variables with that information in the parent, so that the animation
- *   can be implemented with them.
- * - Adds a `is-animation-enabled` CSS class when the element changes, so that the
- *   target (e.g. the indicator) can be animated to its new position.
- * - Removes the `is-animation-enabled` class when the animation is done.
+ * - Adds CSS variables with rect information to the container, so that the indicator
+ *   can be rendered and animated with them. These are kept up-to-date, enabling CSS
+ *   transitions on change.
+ * - Sets an attribute (`data-subelement-animated` by default) when the tracked
+ *   element changes, so that the target (e.g. the indicator) can be animated to its
+ *   new size and position.
+ * - Removes the attribute when the animation is done.
+ *
+ * The need for the attribute is due to the fact that the rect might update in
+ * situations other than when the tracked element changes, e.g. the tracked element
+ * might be resized. In such cases, there is no need to animate the indicator, and
+ * the change in size or position of the indicator needs to be reflected immediately.
  */
-function useSubelementAnimation(
-	subelement?: HTMLElement | null,
+function useAnimatedOffsetRect(
+	/**
+	 * The container element.
+	 */
+	container: HTMLElement | undefined,
```

Comment:
Nitty, but is there a reason why `null` is not a valid type here?

Reply:
Why would you need null? This pleases TypeScript so I didn't give it a second thought. I tend to avoid null whenever possible. Having two void values in JavaScript was a mistake.

Reply:
Probably a habit caused by working with ref callbacks and vanilla APIs like `document.querySelector`, which use `null` to represent "no element".

Happy to keep things as they are, though.